### PR TITLE
Optimize association eager loading for a single model

### PR DESCRIPTION
**Pull request by jturkel**
This change optimizes automatic eager loading when there's only a single model in the auto include context by just doing regular Rails association loading. This bypasses a bunch of extra work that the Rails preloader does. In benchmarks this reduces object allocations by 5% but does not result in a statistically significant throughput change (even though it was faster by about 10% in all of the experiments I ran):

```
Goldiloader: Single model's association
                          2.287k (± 8.5%) i/s -     22.848k in  10.069578s
Goldiloader: Single model's association optimized
                          2.536k (± 7.9%) i/s -     25.194k in  10.004004s

Comparison:
Goldiloader: Single model's association optimized:     2535.6 i/s
Goldiloader: Single model's association:     2287.4 i/s - same-ish: difference falls within error
```

I added a new benchmark as part of this change so I factored out some commonality between the benchmarks and added a script for running all benchmarks.

@fgarces - you're prime

### CODE REVIEW

File: benchmark/single_model_benchmark.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require_relative 'performance_helper'
+
+ForkingBenchmark.ips do |x|
+  x.time = 5
```
Comment (fgarces) [inline benchmark/single_model_benchmark.rb]: is there any reason for 5 and 2 or just because they sounded fine?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ### 3.0.3
 - Optimize association eager loadable checks by caching information on the association's reflection.
+- Optimize association eager loading if we're only eager loading associations for a single model.
```
Comment (fgarces) [inline CHANGELOG.md]: I assume you havent released the gem, otherwise this should go into 3.0.4.

Comment (jturkel) [reply CHANGELOG.md]: Yup. I haven't released 3.0.3 yet.

File: benchmark/single_model_benchmark.rb (hunk 1 above)
Comment (jturkel) [reply benchmark/single_model_benchmark.rb]: They were copied from the benchmark that landed in #77. Looks like these are the defaults for benchmark-ips though so I'll remove them.

**Pull request merged**